test/parallel/test-assert.js

```diff
@@ -655,40 +655,46 @@ try {
                      'Message incorrectly marked as generated');
 }
 
-// Verify that throws() and doesNotThrow() throw on non-function block
-function testBlockTypeError(method, block) {
-  let threw = true;
-
-  try {
-    method(block);
-    threw = false;
-  } catch (e) {
-    assert.strictEqual(e.toString(),
-                       'TypeError: "block" argument must be a function');
-  }
+{
+  // Verify that throws() and doesNotThrow() throw on non-function block
+  const validationFunction = common.expectsError({
+    code: 'ERR_INVALID_ARG_TYPE',
+    type: TypeError
+  });
+
+  const testBlockTypeError = (method, block) => {
+    let threw = true;
+
+    try {
+      method(block);
+      threw = false;
+    } catch (e) {
+      validationFunction(e);
+    }
+
+    assert.ok(threw);
+  };
 
-  assert.ok(threw);
+  testBlockTypeError(assert.throws, 'string');
```

Comment:
@addaleax Tempted to leave this PR as-is just so I can put your suggestion on a list for Code + Learn... Is that unhealthy? :-P